### PR TITLE
Add payment process attach case flow

### PR DESCRIPTION
**Pull request by mdayican**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-864

### Change description ###

add payment handling for  attachToCase callback

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by mdayican** [comment]
> no new tests required?

probably yes. let me see how pipeline goes

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java [hunk 1]
```diff
@@ -48,13 +49,17 @@
 public class AttachCaseCallbackService {
 
     public static final String INTERNAL_ERROR_MSG = "An error occurred";
+    public static final String PAYMENT_ERROR_MSG = "Payment reference can not be processed. Please try again later";
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java]: This constant doesn't seem to be used anywhere.

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java]: NO REVIEW for now

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java [hunk 2]
```diff
@@ -264,18 +281,29 @@ private void attachCaseByCcdId(
...
         );
 
-        log.info("Attached exception record '{}' to case with CCD ID '{}'", exceptionRecordId, targetCaseCcdRef);
+        log.info(
+            "Successfully attach Exception Record with ID {} to case with id {}",
+            theCase.getId(),
+            exceptionRecord.getId()
+        );
+
+        paymentsProcessor.updatePayments(exceptionRecord, theCase.getId());
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java]: Calling payments processor is very unexpected in a method that's supposed to update a case in CCD. It should be done after this method has completed.

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java (hunk 1 above)
Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java]: > NO REVIEW for now

can we mark it somehow so reviewers are aware it's WIP

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java [hunk 3]
```diff
@@ -91,7 +94,17 @@ public AttachCaseCallbackService(CcdApi ccdApi) {
...
             )
+            .peek(result -> processPayment(result, exceptionRecord))
             .getOrElseGet(errors -> Either.left(errors.toJavaList()));
+
+    }
+
+    private void processPayment(Either<List<String>, Map<String, Object>> result, CaseDetails exceptionRecord) {
+        if (result.isRight() && result.get() != null) {
+            Long caseId = Long.valueOf((String) result.get().get(ATTACH_TO_CASE_REFERENCE));
+            log.info("Call update payment processor, caseId {}", caseId);
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java]: there's a log already present in processor `Sending payment update message. ER id: {}`. We are overusing logs :( - we can just extend that one

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/payments/PaymentsPublishingException.java [hunk 4]
```diff
@@ -1,6 +1,10 @@
 package uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.domains.payments;
 
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
 @SuppressWarnings("serial")
+@ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/payments/PaymentsPublishingException.java]: If we good with throwing this exception to UI then implementation with peeks and void methods in payment processor is fine

**Comment by mdayican** [comment]
ticket changed, I will create new PR

**Pull request closed without merging**